src/clients/python/experimental_api_v2/library/httpclient.py

```diff
@@ -107,198 +104,447 @@ def close(self):
-        """Request the inference server to load or reload specified model.
+        if not model_version:
+            request_uri = "v2/models/{}".format(model_name)
+        else:
+            request_uri = "v2/models/{}/versions/{}".format(
+                model_name, model_version)
+        response = self._client_stub.get(request_uri, headers=headers)
+        raise_if_error(response)
+        metadata = json.loads(response.read())
+        return metadata
+
+    def infer(self,
+              inputs,
+              model_name,
+              model_version="",
+              outputs=None,
+              request_id=None,
+              parameters=None,
+              headers={}):
+        """Run synchronous inference using the supplied 'inputs' requesting
+        the outputs specified by 'outputs'.
 
         Parameters
         ----------
-        model_name : str
-            The name of the model to be loaded.
+        inputs : list
+            A list of InferInput objects, each describing data for a input
+            tensor required by the model.
+        outputs : list
+            A list of InferOutput objects, each describing how the output
+            data must be returned. If not specified all outputs produced
+            by the model will be returned using default settings.
+        model_name: str
+            The name of the model to run inference.
+        model_version: str
+            The version of the model to run inference. The default value
+            is an empty string which means then the server will choose
+            a version based on the model and internal policy.
+        request_id: str
+            Optional identifier for the request. If specified will be returned
+            in the response. Default value is 'None' which means no request_id
+            will be used.
+        parameters: dict
+            Optional inference parameters described as key-value pairs.
+        headers: dict
+            Optional dictionary specifying the headers that must be
+            sent alongwith the request
+
+        Returns
+        -------
+        InferResult
+            The object holding the result of the inference, including the
+            statistics.
 
         Raises
         ------
         InferenceServerException
-            If unable to load the model.
+            If server fails to perform inference.
+        """
+        infer_request = {}
+        if request_id:
+            infer_request['id'] = request_id
+        if parameters:
+            infer_request['parameters'] = parameters
+        infer_request['inputs'] = [
+            this_input._get_tensor() for this_input in inputs
+        ]
+        if outputs:
+            infer_request['outputs'] = [
+                this_output._get_tensor() for this_output in outputs
+            ]
+
+        request_body = json.dumps(infer_request)
+        if not model_version:
+            request_uri = "v2/models/{}/infer".format(model_name)
```

Comment:
Are model names guaranteed to only have URL-safe characters?  

Otherwise something like https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote